src/main/java/duke/exception/InvalidCommand.java

```diff
@@ -0,0 +1,30 @@
+package duke.exception;
+
+/**
+ * Represents an exception that will be thrown when the user enters an invalid command.
+ *
+ * @author Terng Yan Long
+ */
+public class InvalidCommand extends DukeException {
+    private final String errorMessage;
+
+    /**
+     * Initialises a new instance of duke.exception.InvalidCommand.
+     *
+     * @param errorMessage The error message that is related to the exception.
+     */
+    public InvalidCommand(String errorMessage) {
+        super(errorMessage);
+        this.errorMessage = errorMessage;
+    }
+
+    /**
+     * Creates an error message to be displayed to the user.
+     *
+     * @return String containing the output message.
+     */
+    @Override
+    public String toString() {
+        return "OOPS!!! " + this.errorMessage;

```

Comment:
I'm really nitpicking here because you've pretty much made like 0 mistakes, but according to a friend, we aren't supposed to use the "this" keyword if possible unless absolutely necessary. Don't know if it's true or not, but I also did use "this" everywhere in my code.